presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetTypeUtils.java

```diff
@@ -108,7 +110,7 @@ public static Type getPrestoType(RichColumnDescriptor descriptor)
             case DOUBLE:
                 return DoubleType.DOUBLE;
             case INT32:
-                return createDecimalType(descriptor).orElse(IntegerType.INTEGER);
+                return createDecimalType(descriptor).orElseGet(() -> createDateType(descriptor).orElse(IntegerType.INTEGER));
```

Comment:
Instead of chaining multiple `orElse` calls, which makes this complicated, let's do `return getInt32Type(descriptor);` and add a new `getInt32Type` method:

```
private static Type getInt32Type(RichColumnDescriptor descriptor)
{
    switch (descriptor.getPrimitiveType().getOriginalType())
    {
        case DECIMAL:
            break;
        case DATE:
            break;
    }
}
```